ecosystem/sep-0012.md

```diff
@@ -503,6 +499,72 @@ Name | Type | Description
+`type` | `string` | (optional) the type of action the customer is being KYCd for. See the [Type Specification](#type-specification) below.
+`status` | `string` | One of the values described in [Provided Field Statuses](#provided-field-statuses)
+
+In order to validate the integrity and provenance of the request, the Anchor MUST include a signature in an additional HTTP Header `X-Stellar-Signature`.
+This Header MUST follow the specification: `X-Stellar-Signature: t=<timestamtp>, h=<host>, s=<base64 signature>` where:
+* __timestamp__ is the current timestamp at the time the callback is sent. This is used to assure the freshness of the request and to prevent this request to be replayed in the future. 
+* __host__ is the host the callback is being sent to. The host will be part of the signature. This is used to prevent malicious wallet to relay the request to another wallet
+* __base64 signature__ is the base64 encoding of the request signature. We explain below how to compute and verify this signature. The signature is computed using the Stellar private key linked to the `SIGNING_KEY` field of the anchor's [`stellar.toml`](sep-0001.md).
+
+It is the wallet's responsability to:
+* Verify the signature using the corresponding Stellar `SIGNING_KEY` field of the anchor's [`stellar.toml`](sep-0001.md).
+* Verify the freshness of the request by comparing the `timestamp` in the request with the current timestamp at the time of the reception and discard every request above a threshold of few seconds.
+* Send a working callback URL to the anchor.
+
+### VERIFY signature
+
+* Check that callback request has `X-Stellar-Signature` header
+* Parse the header and extract:
+  * Key `t`: __timestamp__
+  * Key `h`: __host__
+  * Key `s`: __base64 signature__
+* Verify the request freshness: _current timestamp_ - __timestamp__ < few seconds (2-5)
+* Extract the __body__ of the request
+* Base64 decode the __base64 signature__ to get the __signature__
+* Prepare the payload to verify the signature:
+  * The __timestamp__ (as a string)
+  * The character `.`
+  * The __host__
+  * The character `.`
+  * The body
+* Verify the signature using the correct `SIGNING_KEY`
+
+### COMPUTE signature
+
+* Prepare the callback request body
+  * `id`
+  * `account`
+  * `memo` (optional)
+  * `memo_type` (optional)
+  * `type` (optional)
+  * `status`
+* Prepare the payload to sign:
+  * Current timestamp (as a string)
+  * The character `.`
+  * The wallet host to send the callback request to
+  * The character `.`
+  * The callback request body
+* Compute the signature using the Anchor private seed
+* Base64 encode the signature
+* Build the `X-Stellar-Signature` header:
+  * `X-Stellar-Signature: t=<current timestamp>, h=<wallet host>, s=<base64 encoded signature>`
+
+Anchors will submit POST requests until the user's status changes to `ACCEPTED` or `REJECTED`. If a wallet needs to watch a user's KYC status after that, it will need to set a callback again.
```

Comment:
Can we add this to the start of the request definition so it doesn't get missed?